### DISCUSSION TITLE
File deleted

### DISCUSSION THREAD

**Discussion by wpoely86** [abraunegg/onedrive#2009] [Q&A] [answered]
This morning, a file got deleted by the onedrive sync (2.4.18)

```
jun 17 08:21:28 beedrill onedrive[1975]: Downloading file Documents/OSO/DICT & OSO.pptx ...
jun 17 08:24:47 beedrill onedrive[1975]: [2.9K blob data]
jun 17 08:24:47 beedrill onedrive[1975]: ERROR: Microsoft OneDrive API returned an error with the following message:
jun 17 08:24:47 beedrill onedrive[1975]:   Error Message:    Failure when receiving data from the peer on handle 1C6C470
jun 17 08:24:47 beedrill onedrive[1975]:   Calling Function: download()
jun 17 08:24:47 beedrill onedrive[1975]: ERROR: File download size mis-match. Increase logging verbosity to determine why.
jun 17 08:24:47 beedrill onedrive[1975]: ERROR: File download hash mis-match. Increase logging verbosity to determine why.
jun 17 08:24:47 beedrill onedrive[1975]: INFO: It is most likely that a SharePoint OneDrive API issue is the root cause. Add --disable-download-validation to work around this issue but downloaded data integrity cannot be guaranteed.
jun 17 08:24:47 beedrill onedrive[1975]: ERROR: The local file system returned an error with the following message:
jun 17 08:24:47 beedrill onedrive[1975]:   Error Message:    Documents/OSO/DICT & OSO.pptx: No such file or directory
jun 17 08:24:47 beedrill onedrive[1975]:   Calling Function: applyChangedItem()
jun 17 08:24:47 beedrill onedrive[1975]: Moving Documents/OSO/DICT & OSO.pptx.partial to Documents/OSO/DICT & OSO.pptx
jun 17 08:24:47 beedrill onedrive[1975]: Cannot move item: Documents/OSO/DICT & OSO.pptx: No such file or directory
jun 17 08:24:47 beedrill onedrive[1975]: Cannot upload file changes/creation: Documents/OSO/DICT & OSO.pptx: No such file or directory
jun 17 08:24:47 beedrill onedrive[1975]: Received inotify delete event from operating system .. attempting item deletion as requested
jun 17 08:24:47 beedrill onedrive[1975]: Deleting item from OneDrive: Documents/OSO/DICT & OSO.pptx
jun 17 08:24:49 beedrill onedrive[1975]: Sync with OneDrive is complete
```

What I don't understand is the message: `Received inotify delete event from operating system .. attempting item deletion as requested`. I was not touching the file in any way locally. Maybe tracker3 was trying to index it but that should never sent a delete event.

Any clue what happened here?

**Comment by abraunegg** [comment, marked as answer]
Your operating system sent an inotify event of file deletion.

The onedrive client processed that request.

The inotify kernel system has no way to advise where the event originated from.

Also you are using an old and now obsolete client - please upgrade your client.

**Reply by wpoely86** [reply]
As far as I can see 2.4.19 was released 2 days ago? I will update as soon as it hits the fedora repos.

**Reply by wpoely86** [reply]
Also, this is very weird. A file is on my system, not being used 'actively' in any way. The sync client has a issue with syncing (most likely because I was on a crappy 4G connection at the time) and suddenly there is a inotify event to delete the file.

What is the sync client doing exactly in this case? The file is there (I see in the hourly snapshots). The download fails. Then we already get an error `Error Message:    Documents/OSO/DICT & OSO.pptx: No such file or directory`. While it was there.
Next it says:
```
jun 17 08:24:47 beedrill onedrive[1975]: Moving Documents/OSO/DICT & OSO.pptx.partial to Documents/OSO/DICT & OSO.pptx
jun 17 08:24:47 beedrill onedrive[1975]: Cannot move item: Documents/OSO/DICT & OSO.pptx: No such file or directory
```
Which I also don't understand. Why would the move of `.partial` fail if the file is not there?
